hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/CalciteSqlOptimizer.java

```diff
@@ -339,6 +341,14 @@ private SqlPlan createPlan(
     ) {
         // TODO [sasha] : refactor this.
         SqlNode node = parseResult.getNode();
+        boolean analyze = false;
+        Map<String, String> analyzeOptions = emptyMap();
+        if (node instanceof SqlAnalyzeStatement) {
```

Comment:
this logic is relevant mostly for `else` branch of the next big `if` (selects and DML), however here you do some unpacking. Is ANALYZE intended to be available for other types of plans? Currently it will be silently ignored.

Reply:
the syntax allows ANALYZE only for Select and DML

Reply:
IMO the code might be hard to understand. If this is intended only for SELECT and DML, I would move it there. Also that `else` branch used to cover Select and DML statements, now it will also cover ANALYZE. This should work fine, but as I said IMO it is not very readable.

Reply:
You're right, makes sense to modularize this a bit. I'll look into maybe introducing ANALYZE as a separate plan similarly to Create Job. Might be useful later on if we decide to add some additional logging or potentially ANALYZE EXPLAIN. As far as job name goes,  yes definitely possible, I think most logical way would be to add it as a clause rather than an option e.g.: 

```
ANALYZE
WITH JOB NAME myJob
WITH OPTIONS ()
WITH EXPLAIN -- not in this version, but with the dedicated plan, its going to be easier to introduce this
SELECT * FROM test
```
wdyt?

Reply:
we can do that refactor when implementing https://hazelcast.atlassian.net/browse/HZ-2865